ocaml/tests/test_pool_repository.ml

```diff
@@ -14,101 +14,156 @@
-  let __context = T.make_test_database () in
-  let name_label = "remote" in
-  let name_description = "remote" in
-  let binary_url = "https://repo.example.com" in
-  let source_url = "https://repo-src.example.com" in
-  let gpgkey_path = "" in
-  let ref_remote =
-    Repository.introduce ~__context ~name_label ~name_description ~binary_url
-      ~source_url ~update:true ~gpgkey_path
+  let ref_remote_pool =
+    Repository.introduce_remote_pool ~__context ~name_label:"remote_pool"
+      ~binary_url:binary_url_2 ~name_description:"remote_pool" ~certificate:""
   in
-  let ref_bundle =
-    Repository.introduce_bundle ~__context ~name_label:"bundle"
-      ~name_description:"bundle"
-  in
-  let self = Helpers.get_pool ~__context in
-  Alcotest.check_raises "test_add_bundle_repo"
-    Api_errors.(Server_error (bundle_repo_should_be_single_enabled, []))
-    (fun () ->
-      Xapi_pool.set_repositories ~__context ~self ~value:[ref_remote] ;
-      Xapi_pool.add_repository ~__context ~self ~value:ref_bundle
-    )
+  f __context pool ref_remote ref_bundle ref_remote_pool
 
-let test_add_remote_repo () =
-  let __context = T.make_test_database () in
-  let name_label = "remote" in
-  let name_description = "remote" in
-  let binary_url = "https://repo.example.com" in
-  let source_url = "https://repo-src.example.com" in
-  let gpgkey_path = "" in
-  let ref_remote =
-    Repository.introduce ~__context ~name_label ~name_description ~binary_url
-      ~source_url ~update:true ~gpgkey_path
-  in
-  let ref_bundle =
-    Repository.introduce_bundle ~__context ~name_label:"bundle"
-      ~name_description:"bundle"
-  in
-  let self = Helpers.get_pool ~__context in
-  Alcotest.check_raises "test_add_remote_repo"
-    Api_errors.(Server_error (bundle_repo_should_be_single_enabled, []))
-    (fun () ->
+let test_set_repositories () =
+  on_repositories (fun __context self ref_remote ref_bundle ref_remote_pool ->
+      Xapi_pool.set_repositories ~__context ~self ~value:[ref_remote] ;
       Xapi_pool.set_repositories ~__context ~self ~value:[ref_bundle] ;
-      Xapi_pool.add_repository ~__context ~self ~value:ref_remote
-    )
+      Xapi_pool.set_repositories ~__context ~self ~value:[ref_remote_pool] ;
+      Alcotest.check_raises "test_set_repositories_1"
+        Api_errors.(
+          Server_error
+            ( repo_should_be_single_one_enabled
+            , [Record_util.origin_to_string `bundle]
+            )
+        )
+        (fun () ->
+          Xapi_pool.set_repositories ~__context ~self
+            ~value:[ref_remote; ref_bundle]
+        ) ;
+      Alcotest.check_raises "test_set_repositories_2"
+        Api_errors.(
+          Server_error
+            ( repo_should_be_single_one_enabled
+            , [
+                String.concat ", "
+                  [
+                    Record_util.origin_to_string `bundle
+                  ; Record_util.origin_to_string `remote_pool
+                  ]
+              ]
```

Comment:
[`bundel; `remote_pool] |> List.map Record_util.origin_to_string
And the same for others.

Reply:
I see several errors are using this format, like `Api_errors.pvs_site_contains_running_proxies`, `Api_errors.network_contains_vif` and `Api_errors.network_contains_pif`, would you please help to confirm if this is indeed a problem we should not follow? Thank you!

Reply:
I followed the existing errors and updated it.